MapboxCoreNavigation/RouteController.swift

```diff
@@ -587,6 +605,49 @@ extension RouteController: CLLocationManagerDelegate {
         guard routeProgress.currentLegProgress.currentStepProgress.durationRemaining > RouteControllerMediumAlertInterval else { return }
         checkForFasterRoute(from: location)
     }
+
+    func beginTunnelAnimation(for manager: CLLocationManager, routeProgress: RouteProgress, distanceTraveled: CLLocationDistance) {
+        guard !(manager is SimulatedLocationManager), simulatedLocationManager == nil else { return }
+
+        simulatedLocationManager = SimulatedLocationManager(route: routeProgress.route, distanceTraveled: distanceTraveled)
+        simulatedLocationManager?.delegate = self
+        simulatedLocationManager?.routeProgress = routeProgress
+        
+        let dispatchGroup = DispatchGroup()
```

Comment:
Can you explain the reasoning behind using this dispatch group?

Reply:
@frederoni Sure. 

This ensures we don't encounter any concurrency (dispatch) race between the local `locationManager` and the `simulatedLocationManager`. Basically, when we dispatch using a group, we are re-assured that our local `locationManager` will stop `stopUpdatingLocation` before the `simulatedLocationManager ` starts `startUpdatingLocation`. 

This facilitates a smooth handshake between using our `RouteController.locationManager` and `RouteController.simulatedLocationManager` where we will not have both running at the same time or vice versa.